app/main.py

```diff
@@ -1,2 +1,109 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Hashable
+
+
+@dataclass
+class DictItem:
+    key: Hashable
+    value: Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self._capacity = 8
+        self._load_factor = 2 / 3
+        self._threshold = int(self._capacity * self._load_factor)
+        self._size = 0
+        self._source = [None for _ in range(self._capacity)]
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = hash(key) % self._capacity
+        is_exist = False
```

Comment:
What do you need this variable for? Can't you just increase `_size` when you set it to `True`?